includes/admin/class-wc-rest-payments-reporting-controller.php

```diff
@@ -44,11 +46,27 @@ public function register_routes() {
 	 * @param WP_REST_Request $request The request.
 	 */
 	public function get_payment_activity( $request ) {
-		$wcpay_request = Get_Reporting_Payment_Activity::create();
-		$wcpay_request->set_date_start( $request->get_param( 'date_start' ) );
-		$wcpay_request->set_date_end( $request->get_param( 'date_end' ) );
+		$wcpay_request                = Get_Reporting_Payment_Activity::create();
+		$date_start_in_store_timezone = $this->format_date_to_wp_timezone( $request->get_param( 'date_start' ) );
+		$date_end_in_store_timezone   = $this->format_date_to_wp_timezone( $request->get_param( 'date_end' ) );
```

Comment:
It looks like this code is undoing logic in `Transactions_List` request class:

https://github.com/Automattic/woocommerce-payments/blob/bebe14e87eb2a4604c82dd212221efc1854302c1/includes/core/server/request/class-list-transactions.php#L80-L81

Should we fix it at the source, i.e. delete that code, rather than reversing it here?

> whereas those on the transactions report pages with filter ( used for landing pages ),used [an adjustment](https://github.com/Automattic/woocommerce-payments/blob/bebe14e87eb2a4604c82dd212221efc1854302c1/includes/core/server/request/class-list-transactions.php#L80) to match the site settings.

I looked at the other similar request classes and didn't see any doing this timezone adjustment. If `Transactions_List`  is inconsistent then we should fix that. We just need to find out:

1. If anyone wants/expects special / current behaviour of `Transactions_List`, i.e. convert to site timezone.
1. If there is explanation or rationale in GitHub history for this change. 

I started digging in git-blame to find out when this conversion was added. It looks like it was long ago, it was there when these pagination controllers were implemented for previous integration/platform project:

- [WCPay as a Platform: Phase 1](https://github.com/Automattic/woocommerce-payments/pull/5321#top)
  - [Dev/paginated request class](https://github.com/Automattic/woocommerce-payments/pull/5281/files#top)

So 🤞🏻  maybe it's just an old oversight/bug?

Reply:
I agree that fixing the main cause ( option (a) is the ideal solution, compared to adding another workaround. Since we have less time now, it would be good to go with option b for now , and create a separate issue where we check 
* Why was the timezone adjustment added to `Transactions_List`
* Downstream implications to other applications / workflows if we remove this adjustment
* Removing time adjustment proposed in the current PR and any other similar changes done.

Reply:
Just noting that we don't _have_ to ship an intermediate patch (aka "b"). Yes there is urgency to wrap up this project, but there is no hard deadline. If we are adding follow up issues, then that undermines this as a way to wrap up the project.

That said I'm not going to stand in the way of a temporary/short term patch if we're confident it's safe. Logging an issue to investigate fix "a" properly is a bottom line for me … but I am concerned it will languish given other pressures.